flink-runtime/src/main/java/org/apache/flink/runtime/state/SharedStateRegistryImpl.java

```diff
@@ -137,8 +146,8 @@ public StreamStateHandle registerReference(
         return entry.stateHandle;
     }
 
-    @Override
-    public void unregisterUnusedState(long lowestCheckpointID) {
+    private void unregisterState(
+            long lowestCheckpointID, Consumer<Long> markCheckpointInUseAction) {
```

Comment:
I think it makes sense to cover the logic of computing `markCheckpointInUseAction` by some unit tests.

It could be possible if `unregisterUnusedState` returns this set as suggested in another comment.

Reply:
I tested  the `markCheckpointInUseAction` in `CheckpointsCleanerTest`.